src/Statement/StatementListChanger.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement;
+
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * A collection of functions to manipulate StatementList objects.
+ *
+ * @license GPL-2.0+
+ * @author Thiemo Mättig
+ */
+class StatementListChanger {
```

Comment:
I'm very open for suggestions.

I definitely do not want this to be a set of static methods. This is something I would have done a few years ago. I learned. :-)

A builder does not work because I definitely want this to manipulate the original object, not create new ones.

Passing the StatementList via the constructor instead of each method does not change much, on the one hand. On the other hand it makes it much harder to replace this implementation with an other one. You can not simply make this an interface and inject an other implementation. You do not know the StatementList object on construction time. This approach would need a factory.

In conclusion I still think my current approach is the best.

> the name should indicate what kind of extra operations these are.

What do you mean with "extra operations"? What is not obvious enough from the method names?